db/migrate/20181105061336_cast_key_types_and_values_in_parameters.rb

```diff
@@ -0,0 +1,46 @@
+class CastKeyTypesAndValuesInParameters < ActiveRecord::Migration[5.2]
+  def up
+    User.as_anonymous_admin do
```

Comment:
in migration, user does not have to exist yet (migration on initial db), if we need top skip permission check, then we need to wrap it with `Parameter.skip_permission_check { ... }`

Reply:
also we should probably disable auditing in this migration, `Parameter.without_auditing { ... }` should do, it created many audit records in my db under internal admin

Reply:
If I run migration with `Parameter.skip_permission_check` on development setup, records are failing  on save with validation messages. With `User.as_anonymous_admin`,  save runs properly without validation failures for records.


```

-- Failed to update param(36): #<HostParameter id: 36, name: "subname", value: "test purpose", reference_id: 3, created_at: "2018-08-25 18:14:12", updated_at: "2018-11-05 06:53:07", type: "HostParameter", priority: 70, hidden_value: false, key_type: "string">
-- Error: ["You don't have permission edit_params with attributes that you have specified or you don't have access to specified organizations or locations"]
-- Failed to update param(37): #<HostParameter id: 37, name: "kt_activation_keys", value: "test1", reference_id: 3, created_at: "2018-08-25 18:14:13", updated_at: "2018-11-05 06:53:07", type: "HostParameter", priority: 70, hidden_value: false, key_type: "string">
-- Failed to update param(11): #<GroupParameter id: 11, name: "fdgfd", value: "fdgdfg", reference_id: 6, created_at: "2018-04-26 13:00:59", updated_at: "2018-11-05 07:03:45", type: "GroupParameter", priority: 60, hidden_value: false, key_type: "string">
-- Error: ["Hostgroup can't be blank"]
-- Failed to update param(30): #<DomainParameter id: 30, name: "Tp1", value: "tp1", reference_id: 1, created_at: "2018-05-27 10:51:44", updated_at: "2019-01-14 13:21:57", type: "DomainParameter", priority: 30, hidden_value: false, key_type: "string">
-- Error: ["Domain can't be blank"]
```